lib/assets/chart_builder/main.js

```diff
@@ -0,0 +1,223 @@
+  }
+
+  function setDataOptions(options) {
+    const selectElem = ctx.root.querySelector(`[name="data"]`);
+    selectElem.innerHTML = "";
+    createOptions(selectElem, Object.keys(options))
+    const selected = selectElem.value;
+    updateDataInfoBox(options);
+    ctx.pushEvent("update_field", { field: "data", value: selected });
+  }
+
+  function setAxisOptions(options) {
+    for (const elem of hasType) {
+      const selectElem = ctx.root.querySelector(`[name="${elem}"]`);
+      selectElem.disabled = false;
+      selectElem.innerHTML = "<option value=''></option>";
+      createOptions(selectElem, options)
+      elem === "color" ? selectElem.selectedIndex = 0 : selectElem.selectedIndex = 1;
+      const selected = selectElem.value;
+      ctx.pushEvent("update_field", { field: elem, value: selected });
+    }
+    setTypes();
+  }
+
+  function setTypes() {
+    for (const elem of types) {
+      const selectElem = ctx.root.querySelector(`[name="${elem}"]`);
+      selectElem.selectedIndex = 0;
+      const selected = selectElem.value;
+      elem === "color_type" ? selectElem.disabled = true : selectElem.disabled = false;
+      ctx.pushEvent("update_field", { field: elem, value: selected });
+    }
+  }
+
+  function maybeReset( {name, value} ) {
+    const elem = ctx.root.querySelector(`[name="${name}_type"]`)
+    value === "" ? resetField(elem) : elem.disabled = false;
+  }
+
+  function missingData() {
+    for (const name of hasType.concat(types)) {
+      const elem = ctx.root.querySelector(`[name="${name}"]`);
+      resetField(elem);
+    }
+    updateDataInfoBox();
+  }
+
+  function resetField(elem) {
+    elem.value = "";
+    elem.disabled = true;
+    ctx.pushEvent("update_field", { field: elem.name, value: "" });
+  }
+
+  function createOptions(selectElem, options) {
+    for (const option of options) {
+      const element = document.createElement("option");
+      element.innerText = option;
+      selectElem.append(element);
+    }
+  }
+
+  function updateInfoBox(dep) {
+    const infoBox = ctx.root.querySelector("#info-box");
+
+    if (dep) {
+      infoBox.classList.remove("hidden");
+      infoBox.textContent = `To successfully build charts, you need to add the following dependency:\n\n    ${dep}`;
+    } else {
+      infoBox.classList.add("hidden");
+    }
+  }
+
+  function updateDataInfoBox(data = null) {
+    const infoBox = ctx.root.querySelector("#data-info-box");
+    if (data) {
+      infoBox.classList.add("hidden")
+    } else {
+      infoBox.classList.remove("hidden");
+      infoBox.textContent =
+  `To successfully plot graphs, you need at least one dataset available.
+  The dataset needs to be a map of series, for exemple:
+
+      my_data = %{
+        a: [89, 124, 09, 67, 45],
+        b: [12, 45, 67, 83, 32]
+      }
+
+  Or using Explorer:
+      iris = Explorer.Datasets.iris() |> Explorer.DataFrame.to_map()
+
+  All datasets declared above a chart builder will be available on it.`;
```

Comment:
```suggestion
The dataset needs to be a map of series, for example:

      my_data = %{
        a: [89, 124, 09, 67, 45],
        b: [12, 45, 67, 83, 32]
      }

Or using Explorer:

      iris = Explorer.Datasets.iris() |> Explorer.DataFrame.to_map()

Chart builder is available for all datasets above.`;
```

Reply:
> Chart builder is available for all datasets above

I'm wondering if it makes sense to say this, given it's the core principle in Livebook in general.